pkg/apis/metal3/v1alpha1/baremetalhost_types.go

```diff
@@ -740,6 +745,26 @@ func (host *BareMetalHost) OperationMetricForState(operation ProvisioningState)
 	return
 }
 
+// GetImageChecksum returns the hash value and its algo.
+func (host *BareMetalHost) GetImageChecksum() (string, string) {
+	hashValue := strings.ToLower(strings.TrimSpace(host.Spec.Image.Checksum))
+	hashAlgo := strings.ToLower(strings.TrimSpace(host.Spec.Image.ChecksumType))
+
+	if hashValue == "" {
+		// Return empty if checksum is not provided
+		return "", ""
+	}
+	if hashAlgo == "" {
+		// If only checksum is specified. Assume type is md5
+		return hashValue, "md5"
+	}
+	switch hashAlgo {
+	case "md5", "sha256", "sha512":
+		return hashValue, hashAlgo
```

Comment:
We should report an explicit error here instead of appearing to ignore the inputs. I think that means this method needs to return an error in addition to the 2 strings.